### PR TITLE
Added HTTP Retrying logic.

### PR DESCRIPTION
**Pull request by codezizo**
- Added retry functionality to http_get and http_post.
- Added set_http_retry_params which allows you to configure retry functionality.
- Removed HTTPError as it was not useful.
- Added relevant tests.
- Added retrying dependency.

### CODE REVIEW

File: requirements_test.txt [hunk 1]
```diff
@@ -2,3 +2,5 @@ mypy==0.770
 pytest==5.4.1
 pytest-flake8==1.0.4
 pytest-localserver==0.5.0
+requests==2.23.0
+retrying==1.3.3
```
[the remark this answers is not in the document]
Comment (codezizo) [reply requirements_test.txt:6]: So test_api.py also requires these dependencies. So I believe I need them here as well?

File: src/dicomweb_client/api.py [hunk 2]
```diff
@@ -7,6 +7,8 @@
 import xml.etree.ElementTree as ET
 from collections import OrderedDict
 from io import BytesIO
+from http import HTTPStatus
+import retrying
```
[the remark this answers is not in the document]
Comment (codezizo) [reply src/dicomweb_client/api.py]: Thanks for the explanation! Done :)

File: requirements_test.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (hackermd) [reply requirements_test.txt:6]: The idea is that `requirements_test.txt` just defines the requirements in addition to those of the library, which are defined in `setup.py`.

Tests are run as follows (see also [.travis.yml](https://github.com/MGHComputationalPathology/dicomweb-client/blob/master/.travis.yml)):

```none
pip install -r requirements_test.txt
pip install .
pytest
```

[the remark this answers is not in the document]
Comment (codezizo) [reply requirements_test.txt:6]: Gotcha! Thanks!

File: setup.py [hunk 3]
```diff
@@ -60,9 +60,10 @@
         'pytest>=5.0',
         'pytest-localserver>=0.5',
         'pytest-flake8>=1.0',
+        'requests>=2.18',
+        'retrying>=1.3.3',
     ],
     install_requires=[
         'pydicom>=1.0',
```
Comment (hackermd) [inline setup.py:67]:
```suggestion
        'pydicom>=1.0',
        'requests>=2.18',
        'retrying>=1.3.3',
```

**Comment by codezizo** [comment]
Great thanks!

**Pull request merged**